modules/gateway/gateway.go

```diff
@@ -51,94 +53,53 @@ type Gateway struct {
 
 // Address returns the NetAddress of the Gateway.
 func (g *Gateway) Address() modules.NetAddress {
-	counter := g.mu.RLock()
-	defer g.mu.RUnlock(counter)
+	id := g.mu.RLock()
+	defer g.mu.RUnlock(id)
```

Comment:
yeah, this caused a naming conflict for me once, where I was already using `id` as an `rpcID`.